### ISSUE TITLE
Sharing a page on Facebook or Twitter

### ISSUE THREAD

**Issue by Brume-AL** [atari-legend/atari-legend#144]
Would it be possible to generate a thumbnail when we share a page on social networks?

At the moment, sharing an AL page on FB looks like this:

![Maupiti Island](https://user-images.githubusercontent.com/12120244/118260445-9ca26980-b4b2-11eb-9121-0e3f85cde630.png)

Wouldn't it be better to share a picture from the game as a thumbnail, like this one? https://www.atarilegend.com/storage/images/game_screenshots/836.png

Thanks in advance.

**Comment by nguillaumin** [comment]
I tried to make it a bit better when sharing Game, Interview, Review and Article pages, however I can't test not being on Facebook or Twitter. It's deployed on the development site, could you try sharing a URL from there and let me know if it works better?

**Comment by Brume-AL** [comment]
It doesn't seem to work on Twitter nor Facebook.

Facebook:
![Capture d’écran 2021-05-14 162028](https://user-images.githubusercontent.com/12120244/118284214-78558580-b4d0-11eb-9874-2236f4e59887.png)

Twitter (card validator):
![Capture d’écran 2021-05-14 162059](https://user-images.githubusercontent.com/12120244/118284225-7b507600-b4d0-11eb-9644-3b2ef5089379.png)

**Comment by nguillaumin** [comment]
Thanks, I think I see. I updated the JSON-LD metadata but I should have updated the OpenGraph HTML tags... I'll have another look.

**Comment by nguillaumin** [comment]
I changed it, please have a look again. I found my old Twitter credentials and it seems to work, although sometimes the images don't show up, I'm not sure why. The image URL is correct as far as I can tell...

**Comment by Brume-AL** [comment]
Seems to work fine, perfect Nicolas!

Do you use a random system for the thumbnail?

Twitter
![Twitter](https://user-images.githubusercontent.com/12120244/118307753-66cda700-b4eb-11eb-97cc-eaa8dce21369.png)

Facebook
![Facebook](https://user-images.githubusercontent.com/12120244/118307759-69300100-b4eb-11eb-937d-7a139c4862bc.png)

**Comment by nguillaumin** [comment]
Thanks for testing. Yet it's picking a random screenshot for the game, we use that in various places like in search results because there's no concept of a "main" screenshot in the database, and there's not really a "first" one as well since they cannot be ordered in CPANEL. So the first one is the first one that was added in CPANEL... But I could switch to that if you prefer.

**Comment by stgraveyard** [comment]
Awesome stuff!

**Comment by Brume-AL** [comment]
You will think I'm very demanding, but would it be possible to have the same thing for the menus? 🙏

**Comment by nguillaumin** [comment]
Sure, however one can only share the page for a menuset, not a specific menu (since there's no page for individual menus). Shall I just pick a random screenshot from one of the disk?

**Comment by nguillaumin** [comment]
I have implemented it on dev using a random screenshot.

**Comment by Brume-AL** [comment]
Yes, a random screenshot can do the job, thanks Nico 👍👍

**Comment by Brume-AL** [comment]
Seems to work great 🙂

![Fuzion](https://user-images.githubusercontent.com/12120244/118370038-20894e00-b5a6-11eb-8e79-b9a1c2822048.png)

The only cases where no image of menu is displayed is when there are a lot of missing menus: 

![CID](https://user-images.githubusercontent.com/12120244/118370044-24b56b80-b5a6-11eb-9ea8-42426f3c19d7.png)

**Comment by nguillaumin** [comment]
I fixed a but where it would only pick a random screenshot from the current page of menus being displayed rather than the whole set, however that was not the cause for "Criminals In Disguise" since there are screenshots on the first page. It looks more that Facebook has cached something because even the description is wrong. Possibly if you tried to link to that set earlier on and it got cached? In any case not much I can do, I checked that the metadata is correct in the page...
